program/rust/src/tests/test_ema.rs

```diff
@@ -148,3 +316,19 @@ struct OutputRecord {
     twap:   i64,
     twac:   i64,
 }
+
+fn populate_instruction(instruction_data: &mut [u8], price: i64, conf: u64, pub_slot: u64) {
+    let mut cmd = load_mut::<UpdPriceArgs>(instruction_data).unwrap();
+    cmd.header = OracleCommand::UpdPrice.into();
+    cmd.status = PC_STATUS_TRADING;
+    cmd.price = price;
+    cmd.confidence = conf;
+    cmd.publishing_slot = pub_slot;
+    cmd.unused_ = 0;
+}
+
+fn add_publisher(price_account: &mut AccountInfo, publisher_key: &Pubkey, index: usize) {
+    let mut price_data = load_checked::<PriceAccount>(price_account, PC_VERSION).unwrap();
+    price_data.num_ = (index + 1) as u32;
+    price_data.comp_[index].pub_ = *publisher_key;
+}
```

Comment:
This function doesn't need to know `index` since you can use the. current `num_` as `index`